src/test/metric_config_test.cpp

```diff
@@ -0,0 +1,257 @@
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    const auto& metricConfig = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig.metricsEnabled, true);
+    ASSERT_EQ(metricConfig.endpointsPath, "/newmetrics");
+    ASSERT_EQ(metricConfig.requestSuccessGrpcPredict, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelReady, true);
+    ASSERT_EQ(metricConfig.requestSuccessGrpcModelReady, false);
+}
+
+TEST_F(MetricsConfigTest, InitOnce) {
+    SetUpConfig(modelMetricsChangedConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+    TearDown();
+    SetUp();
+    SetUpConfig(modelDefaultConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+
+    status = manager.loadConfig(configFilePath);
+
+    const auto& metricConfig2 = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig2.metricsEnabled, true);
+    ASSERT_EQ(metricConfig2.endpointsPath, "/newmetrics");
+    ASSERT_EQ(metricConfig2.requestSuccessGrpcPredict, true);
+    ASSERT_EQ(metricConfig2.requestFailRestModelReady, true);
+    ASSERT_EQ(metricConfig2.requestSuccessGrpcModelReady, false);
+}
+
+TEST_F(MetricsConfigTest, MetricsAllEnabledTest) {
+    SetUpConfig(modelMetricsAllEnabledConfig);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_TRUE(status.ok());
+
+    const auto& metricConfig = manager.getMetricConfig();
+    ASSERT_EQ(metricConfig.metricsEnabled, true);
+    ASSERT_EQ(metricConfig.endpointsPath, "/metrics");
+    ASSERT_EQ(metricConfig.requestFailGrpcModelInfer, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelInfer, true);
+    ASSERT_EQ(metricConfig.requestFailRestModelMetadata, true);
+}
+
+TEST_F(MetricsConfigTest, MetricsBadEndpoint) {
+    SetUpConfig(modelMetricsBadEndpoint);
+    std::filesystem::copy("/ovms/src/test/dummy", modelPath, std::filesystem::copy_options::recursive);
+    createConfigFileWithContent(ovmsConfig, configFilePath);
+    ConstructorEnabledModelManager manager;
+
+    auto status = manager.loadConfig(configFilePath);
+    ASSERT_EQ(status, StatusCode::INVALID_METRICS_ENDPOINT) << status.string();
+}
+
+class ModelMetricReporterTest : public ::testing::Test {};
+
+TEST_F(ModelMetricReporterTest, MetricReporterConstructorTest) {
+    MetricRegistry registry;
+    ModelMetricReporter reporter1 = ModelMetricReporter(nullptr, nullptr, "example_pipeline_name", 1);
```

Comment:
Here you copy metric reporter. Metric reporter should not be copied - I forgot to disable it by deleting copy constructor. (this is a bug discovered by klocwork)

Use:
```
ModelMetricReporter reporter1(nullptr, nullptr, "example_pipeline_name", 1);
```